modules/alloy/src/main/ts/ephox/alloy/positioning/mode/ContentAnchorCommon.ts

```diff
@@ -20,6 +23,18 @@ const capRect = (left: number, top: number, width: number, height: number): Opti
     newTop = 0;
     newHeight = height + top;
   }
+
+  // Try to prevent the context toolbar from getting below the editor toolbar if we know the bounds of the viewport.
```

Comment:
Ideally, this code wouldn't mention "editor toolbar" or 'context toolbar" in a comment (it does above as well). It should only mention the values it understands (parameters passed in).

Also, `capRect` should have some unit tests. If only to give sample input -> output mappings.

Reply:
Seems like a good idea.